### PR TITLE
Blacklist update

### PR DESCRIPTION
**Pull request by Eugene-hu**
Improve the blacklist on both advanced and template server

### CODE REVIEW

**Review by shibshib — APPROVED** [review]

File: bittensor/_neuron/text/template_server/run.py [hunk 1]
```diff
@@ -101,15 +104,61 @@ def blacklist(pubkey:str, request_type:bittensor.proto.RequestType) -> bool:
...
+            else:
+                timecheck[pubkey] = current_time
                 return False
+
+        # Black list or not
+        try:
+            registration_check()
+
+            stake_check()
+
```
Comment (isabella618033) [inline bittensor/_neuron/text/template_server/run.py:154]: hmmm no time check needed for template_server?

File: bittensor/_neuron/text/template_server/run.py [hunk 2]
```diff
@@ -101,15 +104,61 @@ def blacklist(pubkey:str, request_type:bittensor.proto.RequestType) -> bool:
...
+            validator_check()
             
+            return False
+
+        except Exception as e:
```
Comment (isabella618033) [inline bittensor/_neuron/text/template_server/run.py:159]: the test would not bug you but ala might hahaha

**Review by isabella618033 — APPROVED** [review]
amazing job!

File: bittensor/_neuron/text/template_server/run.py (hunk 2 above)
Comment (Eugene-hu) [reply bittensor/_neuron/text/template_server/run.py:159]: haha this is to catch exceptions from the tests, so tests can run in sequential. I think it should be fine?

File: bittensor/_neuron/text/template_server/run.py (hunk 1 above)
Comment (Eugene-hu) [reply bittensor/_neuron/text/template_server/run.py:154]: Still debating if timecheck  is necessary, since multiprocessing would break the timechecks

Comment (isabella618033) [reply bittensor/_neuron/text/template_server/run.py:154]: But then the core validator is not multiprocessesd on sending dendrite calls rn.

File: bittensor/_neuron/text/template_server/run.py (hunk 2 above)
Comment (shibshib) [reply bittensor/_neuron/text/template_server/run.py:159]: ugh. will allow.

**Pull request merged**